File: app/src/main/java/com/bearcave/passageplanning/data/FilesManager.java
package com.bearcave.passageplanning.data;


import android.content.Context;
import android.os.Environment;

import com.bearcave.passageplanning.data.database.DatabaseManager;

import java.io.File;

public class FilesManager {
    private final String DIRECTORY ;
    private final String DATABASE = "database.db";

    private final Context context;


    public FilesManager(Context context) {
        this.context = context;
        DIRECTORY = Environment.getExternalStorageDirectory() + File.separator + "Android/data"+ File.separator + context.getPackageName();
//        context.getDir(DIRECTORY, Context.MODE_PRIVATE);
        createDatabase();
    }

    public DatabaseManager createDatabase(){
        return new DatabaseManager(context,  DATABASE);
    }
}

File: app/src/main/java/com/bearcave/passageplanning/data/FilesManager.kt
package com.bearcave.passageplanning.data


import android.content.Context
import com.bearcave.passageplanning.data.database.DatabaseManager


class FilesManager(private val context: Context) {
    //private val DIRECTORY: String = Environment.getExternalStorageDirectory().toString() + File.separator + "Android/data" + File.separator + context.packageName
    private val DATABASE = "database.db"


    init {
        createDatabase()
    }

    fun createDatabase(): DatabaseManager {
        return DatabaseManager(context, DATABASE)
    }
}
